controllers/configurationpolicy_controller.go

```diff
@@ -511,6 +511,25 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 
 	log.V(2).Info("Processing the object templates", "count", len(plc.Spec.ObjectTemplates))
 
+	// Parse and fetch details from each object in each objectTemplate, and gather namespaces if required
+	var templateObjs []objectTemplateDetails
+	var selectedNamespaces []string
+	var objTmplStatusChangeNeeded bool
+
+	templateObjs, selectedNamespaces, objTmplStatusChangeNeeded, err = r.getObjectTemplateDetails(plc)
```

Comment:
I was starting to wonder if I introduced a bug somewhere... 😞 

So the issue with moving it here is that if one of the object details fetched is templated, it won't be resolved yet when this is called here.

Reply:
Ah nevermind, sorry--I found the loop you're referring to. But my template comment about moving this code back to its original location I think still applies.

Reply:
I think the best approach is loop over the object templates once for policy template processing. Then call `getObjectTemplateDetails` and then loop over the object templates again now that the policy templates are processed.

Reply:
With the new update it will have a loop for resolving templates, then do the `getObjectTemplateDetails`, then have another loop for the other stuff.